app/src/server/emails/templates/PaymentFailed.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from "react";
+import { Button, EmailLayout, Heading, InvoiceDescription, Text } from "./layout";
+import { Link } from "@react-email/components";
+import { demoDescription } from "./layout/DemoData";
+
+interface Props {
+  projectName: string;
+  amount: string;
+  paymentMethodsLink: string;
+  invoicesLink: string;
+  projectLink: string;
+  billingPeriod: string;
+  description?: Record<string, string>[];
+}
+
+const PaymentFailed = ({
+  projectName,
+  amount,
+  paymentMethodsLink,
+  invoicesLink,
+  projectLink,
+  billingPeriod,
+  description,
+}: Props) => {
+  const previewText = `Add a payment method. OpenPipe Usage ${billingPeriod}.`;
+
+  return (
+    <EmailLayout previewText={previewText}>
+      <Heading>Payment required</Heading>
+      <Text>Avoid any interruption of service by adding a payment method.</Text>
```

Comment:
Yes, it looks good!
![failed](https://github.com/OpenPipe/OpenPipe/assets/28661894/99ce4868-44ac-4ff2-9ae8-1e1c4894a214)
I added a text because I feel it is not explicit enough. I also changed "Add a payment method" to "Update a payment method". In most cases, it will fail if a user already has a connected payment method.

Reply:
Looks good! I would say "updating your payment method" instead of "updating a payment method".

Reply:
Changed.